index.bs

```diff
@@ -3830,6 +3862,804 @@ opened</dfn> steps given |window|, |type| and |message|.
+redirect matches that of the request that initiated it.
+
+#### The network.RequestData type #### {#type-network-RequestData}
+
+[=Remote end definition=] and [=local end definition=]
+
+<pre class="cddl local-cddl">
+network.RequestData = {
+    request: network.Request,
+    url: text,
+    method: text,
+    headers: [*network.Header],
+    cookies: [*network.Cookie],
+    headersSize: js-uint,
+    bodySize: js-uint / null,
+    timings: network.FetchTimingInfo,
+};
+</pre>
+
+The <code>network.RequestData</code> type represents an ongoing network request.
+
+<div algorithm>
+
+To <dfn>get the request data</dfn> given |request|:
+
+1. Let |request id| be request's [=request id=].
+
+1. Let |url| be the result of running the [=URL serializer=] with |request|'s
+   [=request/URL=].
+
+1. Let |method| be |request|'s [=request/method=].
+
+1. Let |body size| be 0.
+
+1. Let |body| be request's [=request/body=].
+
+1. If |body| is a [=byte sequence=], let |body size| be the length of that
+   sequence. Otherwise, if |body| is a [=/body=] then let |body size| be that
+   body's <a spec=fetch>length</a>.
+
+   TODO: Should we handle the case where the body length is not yet known?
+
+1. Let |headers size| be the size in bytes of |request|'s [=request/headers list=] when
+   serialized as mandated by [[HTTP11]].
+
+   Note: For protocols which allow header compression, this is the compressed
+   size of the headers, as sent over the network.
+
+1. Let |headers| be an empty list.
+
+1. Let |cookies| be an empty list.
+
+1. For each (|name|, |value|) in |request|'s [=request/headers list=]:
+
+  1. Append the result of [=get a header=] with |name| and |value| to |headers|.
+
+  1. If |name| is a [=byte-case-insensitive=] match for "<code>Cookie</code>" then:
+
+    1. For each |cookie| in the user agent's cookie store that are included in
+       |request|:
+
+       Note: [[COOKIES]] defines some baseline requirements for which cookies in
+       the store can be included in a request, but user agents are free to
+       impose additional constraints.
+
+       1. Append the result of [=get a cookie=] given |cookie| to |cookies|.
+
+1. Let |timings| be [=get the fetch timings=] with |request|.
+
+1. Return a map matching the <code>network.RequestData</code> production, with
+   the <code>request</code> field set to |request id|, <code>url</code> field
+   set to |url|, the <code>method</code> field set to |method|, the
+   <code>headers</code> field set to |headers|, the |cookies| field set to
+   |cookies|, and the <code>headersSize</code> field set to |headers size|,
+   the <code>bodySize</code> field set to |body size|, and the
+   <code>timings</code> field set to |timings|.
+
+</div>
+
+#### The network.ResponseContent type #### {#type-network-ResponseContent}
+
+[=Remote end definition=] and [=local end definition=]
+
+<pre class="cddl local-cddl">
+network.ResponseContent = {
+    size: js-unit / null
```

Comment:
Can `size` actually be null? https://fetch.spec.whatwg.org/#fetch-timing-info-decoded-body-size defaults to 0 so it doesn't seem like it.